### PR TITLE
ManagementGroups API update (2017-11-01-preview)

### PR DESCRIPTION
**Pull request by grzegorzzygmunt**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+          "GetManagementGroupsWithExpandAndResurse": {
+            "$ref": "./examples/getManagementGroupWithExpandAndRecurse.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "ManagementGroups"
+        ],
+        "operationId": "ManagementGroups_Create",
```
Comment (jhendrixMSFT) [inline specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: For methods like this the convention is to name them CreateOrUpdate.

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+            "$ref": "#/parameters/GroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: response body missing

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "DeleteManagementGroup": {
+            "$ref": "./examples/deleteManagementGroup.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Management/managementGroups/{groupId}/subscriptions/{subscriptionId}": {
```
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Just trying to remember what we discussed - did we say we wont expose GET API through this path?

**Comment by jhendrixMSFT** [comment]
Ping @grzegorzzygmunt, have you had a chance to review @ravbhatnagar feedback?

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json (hunk 1 above)
Comment (grzegorzzygmunt) [reply specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Fixed.

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json (hunk 3 above)
Comment (grzegorzzygmunt) [reply specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Added GET API to make things consistent.

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json (hunk 2 above)
Comment (grzegorzzygmunt) [reply specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Added

**Comment by jhendrixMSFT** [comment]
Latest iteration has some linter errors, can you please take a look?  https://travis-ci.org/Azure/azure-rest-api-specs/jobs/315603525

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json [hunk 4]
```diff
@@ -0,0 +1,708 @@
...
+                    "type": "string",
+                    "description": "The identity of the principal or process that updated the object."
+                },
+                "parent": {
+                    "title": "Parent",
+                    "$ref": "#/definitions/ParentGroupInfo"
+                }
+            }
+        },
+        "ManagementGroupChildInfo": {
```
Comment (ravbhatnagar) [inline specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Please add id, name and type top level properties for this.

**Comment by ravbhatnagar** [comment]
@jhendrixMSFT - good to merge after the above issue raised by me has been fixed.

File: specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json (hunk 4 above)
Comment (grzegorzzygmunt) [reply specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: Removing GET for now to match private preview API release. GET for the subscription under management group will be added in the next version of the API.

Comment (marstr) [reply specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json]: @ravbhatnagar Mind confirming whether or not this is sufficient?

**Comment by marstr** [comment]
Looking for this to be reassigned due to the load of PRs I'm taking on in regards to the December refactoring.

**Comment by jhendrixMSFT** [comment]
@grzegorzzygmunt Looks like the feedback from @ravbhatnagar hasn't been addressed yet.

**Comment by ravbhatnagar** [comment]
This looks good. Please confirm with @grzegorzzygmunt if this is the current shape of the APIs or have there been any changes. From talking with some folks on the Management groups team, there may have been some changes. Good to merge if we get confirmation from @grzegorzzygmunt

**Comment by grzegorzzygmunt** [comment]
This represents the current shape of 2017-11-01-preview API which is in the private preview.
There are changes being made, but those are for new version of API.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jhendrixMSFT** [comment]
@grzegorzzygmunt I think the only remaining thing is to update the [readme](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/readme.md), or was that intentionally omitted?

**Comment by jhendrixMSFT** [comment]
Ping @grzegorzzygmunt do you intend to update the readme?

**Comment by jhendrixMSFT** [comment]
It appears that there are two copies of management.json.
- specification/resources/resource-manager/Microsoft.Management/2017-11-01-preview/management.json
- specification/resources/resource-manager/Microsoft.Management/preview/2017-11-01-preview/management.json
With the new directory structure all preview swaggers should be in a preview directory, so I think the first one should be removed.

**Comment by jhendrixMSFT** [comment]
@amarzavery @mcardosos I'm unsure how to handle this model validation error.  It looks related to issue https://github.com/Azure/oav/issues/155 which has since been closed (fixed?).  Are we supposed to add x-nullable: true to the swagger?

**Comment by mcardosos** [comment]
I tried adding `x-nullable: true` to swagger and that didn't work, also debugged a little bit. The issue is in OAV.

**Comment by jhendrixMSFT** [comment]
@mcardosos thanks for investigating, I've opened issue https://github.com/Azure/oav/issues/198 to track it.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**